frontend/ios/DroidKaigi 2019/Scenes/Announcement/AnnouncementDataSource.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  AnnouncementDataSource.swift
+//  DroidKaigi 2019
+//
+//  Created by 佐々木美穂 on 2019/02/02.
+//
+
+import Foundation
+import UIKit
+import RxCocoa
+import RxSwift
+import ioscombined
+
+final class AnnouncementDataSource: NSObject, UITableViewDataSource {
+    typealias Element = [Announcement]
+    var items: Element = []
+    
+    private var cellHeightsCache = [IndexPath: CGFloat]()
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return items.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: AnnouncementTableViewCell = (tableView.dequeueReusableCell(withIdentifier: "AnnouncementCell", for: indexPath) as? AnnouncementTableViewCell)!
+        let announcement = items[indexPath.row]
+        cell.announcement = announcement
+        return cell
+    }
+}
+
+//extension AnnouncementDataSource: RxTableViewDataSourceType {
+//    public func tableView(_ tableView: UITableView, observedEvent: Event<[Announcement]>) {
+//        Binder(self) { dataSource, element in
+//            dataSource.items = element
+//            tableView.reloadData()
+//            }
+//            .on(observedEvent)
+//    }
+//}
+
+extension AnnouncementDataSource: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: false)
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        cellHeightsCache[indexPath] = cell.frame.height
```

Comment:
You don't need to cache this cell height.  
we need to cache it only when heavy calculation is required. (ex. when TableViewCell have collectionViewCell)